lib/couchbase-orm/json_schema.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require "json"
+require 'json-schema'
+require 'singleton'
+
+module CouchbaseOrm
+
+  module JsonSchema
+
+    class Loader
+      include Singleton
+
+      attr_reader :schemas
+
+      def initialize
+        initialize_schemas(ENV['CB_ORM_JSON_SCHEMA_PATH'])
+        super
+      end
+
+      # only visible for test
+      def reset
+        @schemas = nil
+      end
+
+      def extract_type(entity)
+        return entity[:type] if entity.present?
+        nil
```

Comment:
Why should we emphasis the nil return here ? It's automatic.
I like early return to make the happy path more readable.
BTW, the `present?` check is maybe not the best suitable here... we will discuss it together in next pairing session.

```suggestion
        return unless entity.present?

        entity[:type]
```